proposals/0000-extension-lifecycle-framework.md

```diff
@@ -0,0 +1,142 @@
+
+The following transitions are included:
+ * (does not exist) -> `Experimental`
+ * `Experimental` -> `Stable`
+ * `Experimental` -> `Deprecated`
+ * `Experimental` -> `Legacy`
+ * `Stable` -> `Deprecated`
+ * `Stable` -> `Legacy`
+ * `Legacy` -> `Deprecated`
+ * `Deprecated` -> (does not exist)
+
+These transitions are explicitly excluded:
+ * `Stable` -> (does not exist)
+ * `Legacy` -> (does not exist)
+
+This is a small restriction, such that `Stable` or `Legacy` extensions require a deprecation cycle prior to removal.
+
+The following transitions are possible but left at the discretion of implementors as to be included or not:
+
+ * (does not exist) -> `Deprecated`
+ * (does not exist) -> `Legacy`
+ * (does not exist) -> `Stable`
+ * `Experimental` -> (does not exist)
+ * `Stable` -> `Experimental`
+ * `Legacy` -> `Experimental`
+ * `Legacy` -> `Stable`
+ * `Deprecated` -> `Experimental`
+ * `Deprecated` -> `Stable`
+ * `Deprecated` -> `Legacy`
+
+However, it also seems plausible that new knowledge might from time to time cause a stable extension to once again be considered experimental, e.g. in the face of soundness bugs or subtle interactions with other features. We explicitly allow for more transitions to allow for unforeseen circumstances. Restrictions are places on removal only. With this we aim to balance predictibility and flexibility. The consideration of both is important as extension classifications are first and foremost a communications channel between language designers, compiler developers, and users.
+
+For existing, or future, language editions such as `GHC2021` or `Haskell98`, it is expected that none of the contained extensions would be `Experimental`. However, this proposal does not seek to impose any particular policy on the inclusion of extensions into language editions - the developers and the steering committee are always in the best position to make a decision about a concrete extension and extension set.
+
+Moving an extension through the lifecycle will require a GHC proposal. This will give users a chance to provide their input, it will help catch unexpected interactions (such as an `Stable` extension implying an `Experimental` one, and thus potentially allowing breakage at any  moment), and it provides a way for language implementers and users to clarify their mutual expectations.
+
+### 2.2. Documentation
+
+The user's guide for each extension documents where it is in the extension lifecycle. We expect this to be next to the "Since" and "Implied by" fields in extension documentation. Additionally, the user's guide should provide a history of the extension's sojourn through the various states, and the GHC versions in which it transitioned, for example from being experimental to stable, or from stable to deprecated.
```

Comment:
The user's guide for each extension should state what category the extension belongs to.  We expect this statement to be next to...

Reply:
Made this change, thank you!